src/AsyncWebSocket.cpp

```diff
@@ -132,17 +132,11 @@ size_t webSocketSendFrame(AsyncClient *client, bool final, uint8_t opcode, bool
  *    AsyncWebSocketMessageBuffer
  */
 
-AsyncWebSocketMessageBuffer::AsyncWebSocketMessageBuffer()
-    : _buffer(std::make_shared<std::vector<uint8_t>>(0)) 
-{
-}
-
-AsyncWebSocketMessageBuffer::AsyncWebSocketMessageBuffer(uint8_t* data, size_t size)
+AsyncWebSocketMessageBuffer::AsyncWebSocketMessageBuffer(const uint8_t* data, size_t size)
     : _buffer(std::make_shared<std::vector<uint8_t>>(size)) 
 {
     if (_buffer->capacity() < size) {
-        _buffer.reset();
-        _buffer = std::make_shared<std::vector<uint8_t>>(0);
```

Comment:
The code here was right: it was coming from the original fork. 

This is a safety check in case the buffer cannot be allocated in `_buffer(std::make_shared<std::vector<uint8_t>>(size))` because of not enough heap, then we check that and we reset the buffer to a 0 size to free memory.

The original code from me-no-dev was also doing something similar when creating a `AsyncWebSocketMessageBuffer`.

The logic here: we ask to allocate a `AsyncWebSocketMessageBuffer` buffer of a specific size. If we cannot, then `AsyncWebSocketMessageBuffer` is not allocated at all.

Reply:
that code seems all strange to me,
if, for some reason, `std::vector` won't be able to allocate it will throw `std::bad_alloc`, which in turn will propagate to `std::make_shared`, it won't create half allocated vector where you could check `_buffer->capacity()`, the code will crash anyway.
Also `_buffer.reset();` does not reset vector's capacity, it will release the shared pointer from the object, but that does not make sense if you are going to create a new empty one instead. Just the one that is already there.

Actually the should not be anything at all, if the allocation was successful, then you are good. If not, then code will crash anyway, either in `throw`  or if some miracle happens, later it will corrupt heap when user code start writing in vector's empty data* pointer.

Reply:
Thanks for the explanations! I've misread when using it and did't see the doc link to the allocator saying it will throw. My mistake. So all is good! Thanks for the PR!